### PR TITLE
Replaced Psalm with PHPStan

### PR DESCRIPTION
**Pull request by marioradu05**

### CODE REVIEW

File: src/App/src/Middleware/RememberMeMiddleware.php [hunk 1]
```diff
@@ -20,12 +20,12 @@ class RememberMeMiddleware implements MiddlewareInterface
 {
     #[Inject(
         UserServiceInterface::class,
-        AuthenticationServiceInterface::class,
+        AuthenticationService::class,
```
Comment (alexmerlin) [inline src/App/src/Middleware/RememberMeMiddleware.php]: Same argument as in https://github.com/dotkernel/admin/pull/302#discussion_r1850177058 - we should stick with `AuthenticationServiceInterface`.

File: src/User/src/Entity/UserRememberMe.php [hunk 2]
```diff
@@ -18,7 +18,7 @@ class UserRememberMe extends AbstractEntity
 
     #[ORM\ManyToOne(targetEntity: User::class)]
     #[ORM\JoinColumn(name: 'userUuid', referencedColumnName: 'uuid', nullable: false)]
-    protected User $user;
+    protected User|UserInterface $user;
```
Comment (alexmerlin) [inline src/User/src/Entity/UserRememberMe.php]: Can't we use only `UserInterface` in this entity, like in [UserAvatar](https://github.com/dotkernel/frontend/blob/5.0/src/User/src/Entity/UserAvatar.php#L23)?

File: src/User/src/Repository/UserRepository.php [hunk 3]
```diff
@@ -103,7 +104,7 @@ public function getRememberUser(string $token): ?UserRememberMe
     /**
      * @throws NonUniqueResultException
      */
-    public function findRememberMeUser(User $user, string $userAgent): ?UserRememberMe
+    public function findRememberMeUser(User|UserInterface $user, string $userAgent): ?UserRememberMe
```
Comment (alexmerlin) [inline src/User/src/Repository/UserRepository.php]: Same here - `UserInterface` should be enough.

**Review by alexmerlin — CHANGES_REQUESTED** [review]

File: src/User/src/Entity/UserRememberMe.php (hunk 2 above)
Comment (marioradu05) [reply src/User/src/Entity/UserRememberMe.php]: I've fixed this

File: src/User/src/Repository/UserRepository.php (hunk 3 above)
Comment (marioradu05) [reply src/User/src/Repository/UserRepository.php]: `User` was used everywhere, using `UserInterface` will introduce some phpstan type hinting issues.

**Review by alexmerlin — APPROVED** [review]

**Pull request merged**